reactor-pool/src/main/java/reactor/pool/AbstractPool.java

```diff
@@ -434,15 +440,22 @@ public void request(long n) {
 		/**
 		 * Stop the countdown started when calling {@link AbstractPool#doAcquire(Borrower)}.
 		 */
-		void stopPendingCountdown() {
+		void stopPendingCountdown(boolean success) {
+			if (!timeoutTask.isDisposed()) {
```

Comment:
Assuming timeoutTask was disposed, what should happen with the pending metric?

Reply:
Isn't the pending metric per connection pool?

Reply:
Just trying to understand - so if a borrower was just timed out, its result can be ignored, while the acquisition will actually account for recording the metric. Do I understand it right?

Reply:
`!timeoutTask.isDisposed()` this check is to ensure that if you invoke this method twice for one and the same `Borrower` you will not record twice.

Reply:
I mean `timeoutTask` is disposed here in this method at the end of the method, so we need to guarantee that we record just once.

Reply:
If the task timeout the recording will happen in `run` method. Is that what you had in mind?

Reply:
I now see what is happening. In another comment I expressed my concern about it. Now my question would be - is there no race here? Is the timeoutTask delivered serially with the acquisition or does it run on potentially another thread? In such case, perhaps the flip of the Borrower's AtomicBoolean state would be the trigger to account for either success or failure in the transitioning from pending state?